sound/soc/intel/common/soc-acpi-intel-adl-match.c

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/*
+ * soc-apci-intel-adl-match.c - tables and support for ADL ACPI enumeration.
+ *
+ * Copyright (c) 2020, Intel Corporation.
+ *
+ */
+
+#include <sound/soc-acpi.h>
+#include <sound/soc-acpi-intel-match.h>
+
+static const struct snd_soc_acpi_endpoint single_endpoint = {
+	.num = 0,
+	.aggregated = 0,
+	.group_position = 0,
+	.group_id = 0,
+};
+
+static const struct snd_soc_acpi_adr_device rt711_0_adr[] = {
+	{
+		.adr = 0x000021025D071100,
+		.num_endpoints = 1,
+		.endpoints = &single_endpoint,
+		.name_prefix = "rt711"
+	}
+};
+
+static const struct snd_soc_acpi_link_adr adl_rvp[] = {
+	{
+		.mask = BIT(0),
+		.num_adr = ARRAY_SIZE(rt711_0_adr),
+		.adr_d = rt711_0_adr,
+	},
+	{}
+};
+
+struct snd_soc_acpi_mach snd_soc_acpi_intel_adl_machines[] = {
+	{},
+};
+EXPORT_SYMBOL_GPL(snd_soc_acpi_intel_adl_machines);
+
+/* this table is used when there is no I2S codec present */
+struct snd_soc_acpi_mach snd_soc_acpi_intel_adl_sdw_machines[] = {
+	{
+		.link_mask = 0x1, /* link0 required */
+		.links = adl_rvp,
+		.drv_name = "sof_sdw",
+		.sof_fw_filename = "sof-adl.ri",
+		.sof_tplg_filename = "sof-adl-rt711.tplg",
+	},
+	{},
+};
+EXPORT_SYMBOL_GPL(snd_soc_acpi_intel_adl_sdw_machines);
+
+MODULE_LICENSE("GPL");
+MODULE_DESCRIPTION("Intel Common ACPI Match module");
```

Comment:
shouldn't this mention ADL?

Reply:
@lyakh Good question. This is good ol' copy'n'paste from other modules:

soc-acpi-intel-adl-match.c:56:MODULE_DESCRIPTION("Intel Common ACPI Match module");
soc-acpi-intel-bxt-match.c:90:MODULE_DESCRIPTION("Intel Common ACPI Match module");
soc-acpi-intel-byt-match.c:238:MODULE_DESCRIPTION("Intel Common ACPI Match module");
soc-acpi-intel-cfl-match.c:23:MODULE_DESCRIPTION("Intel Common ACPI Match module");
soc-acpi-intel-cht-match.c:201:MODULE_DESCRIPTION("Intel Common ACPI Match module");
soc-acpi-intel-cml-match.c:314:MODULE_DESCRIPTION("Intel Common ACPI Match module");
soc-acpi-intel-cnl-match.c:68:MODULE_DESCRIPTION("Intel Common ACPI Match module");
soc-acpi-intel-ehl-match.c:25:MODULE_DESCRIPTION("Intel Common ACPI Match module");
soc-acpi-intel-glk-match.c:48:MODULE_DESCRIPTION("Intel Common ACPI Match module");
soc-acpi-intel-hda-match.c:40:MODULE_DESCRIPTION("Intel Common ACPI Match module");

Reply:
@lyakh I kept this as it is, as these are all compiled into snd-soc-acpi-intel-match.ko so the name is accurate. 

OTOH, the modinfo output looks strange:
```
# modinfo snd-soc-acpi-intel-match
filename:       /lib/modules/5.8.10+/kernel/sound/soc/intel/common/snd-soc-acpi-intel-match.ko
description:    Intel Common ACPI Match module
license:        GPL v2
description:    Intel Common ACPI Match module
license:        GPL v2
description:    Intel Common ACPI Match module
license:        GPL v2
description:    Intel Common ACPI Match module
license:        GPL v2
description:    Intel Common ACPI Match module
license:        GPL v2
description:    Intel Common ACPI Match module
license:        GPL v2
description:    Intel Common ACPI Match module
license:        GPL v2
description:    Intel Common ACPI Match module
license:        GPL v2
description:    Intel Common ACPI Match module
license:        GPL v2
description:    Intel Common ACPI Match module
license:        GPL v2
description:    Intel Common ACPI Match module
license:        GPL v2
```

@plbossart Should we clean this up and have the MODULE_* tags in a single file and remove them from all the platform specific variants?

Reply:
Oh! Yes, please, no idea why we need `MODULE_DESCRIPTION()` and `MODULE_LICENSE()` in each of those files, unconditionally linked into a single module

Reply:
yes we should clean this up....
Also IIRC Mark Brown wanted us to make the initial comments more C++, can't recall exactly.
Let's do this in a separate PR.

Reply:
@plbossart I did the modinfo cleanup now: https://github.com/thesofproject/linux/pull/2602
I'll update this PR to match. I want to wait until @libinyang ack's this before merging anyways.